### ISSUE TITLE
Notifications - Extensions

### ISSUE THREAD

**Issue by m-prodan** [bcgov/foi-flow#1168]
* As a Ministry Coordinator
* I want to be notified when an extension has been taken on a file I am assigned 
* so that I can prioritize my workload

**Assumptions & Scope**	
What are the assumptions for this story?

What is IN scope?

What is NOT in scope?

**Acceptance Criteria**	

**Scenario 1: xxxxxx**
* GIVEN ... (you or your condition)
* WHEN ... (what you do) 
* THEN ... (what you see) 

**Scenario 2: xxxxxx**
* GIVEN ... (you or your condition)
* WHEN ... (what you do) 
* THEN ... (what you see) 

**Scenario 3: xxxxxx**
...

**Dependencies? What is the impact of this dependency? (If so, link dependency in the ticket, make it visible in a team´s backlog)**

**Validation Rules? (If yes, list here)**	

**Design**
@xxx - please link the Design here

**Definition of Ready**

1. [ ] Is there a well articulated User Story?
2. [ ] Is there Acceptance Criteria that covers all scenarios (happy/sad paths)?
3. [ ] If there is a user interface, is there a design? 
4. [ ] Does the user story need user research/validation?
5. [ ] Does this User Story needs stakeholder approval?
6. [ ] Design / Solution accepted by Product Owner
7. [ ] Is this user story small enough to be completed in a Sprint? Should it be split?
8. [ ] Are the dependencies known/ understood? (technical, business, regulatory/policy)
9. [ ] Has the story been estimated?

**Definition of Done**

1.    [ ] Passes developer unit tests
2.    [ ] Passes peer code review
3.    [ ] If there's a user interface, passes UX assurance
4.    [ ] Passes QA of Acceptance Criteria with verification in Dev and Test
5.	[ ] Confirm Test cases built and succeeding 
6.	[ ] No regression test failures
7.	[ ] Test coverage acceptable by Product Owner
8.	[ ] Ticket ready to be merged to master or story branch
9.	[ ] Developer to list Config changes/ Update documents and designs
10.	[ ] Can be demoed in Sprint Review 
11.	[ ] Tagged as part of a Release
12.   [ ] Feature flagged if required
13.  [ ] Change Management activities done?

**Comment by m-prodan** [comment]
Notifications for extensions AC drafted - thoughts welcome @arielleandrews - I was thinking we keep it simple with the message since they are just an alert. I also thought it makes sense that only assigned Ministry users will get the notification - so any IAO watcher will not be notified, which feels consistent with current practice anyways.

**Comment by arielleandrews** [comment]
Nice one Matt! I would say in the messages we don't need to include "FOI Request" before the number, I think the file number is sufficient and will keep it concise and less cluttered.

For Scenario 4, I'm wondering if this needs to be a notification or should just be a system generated comment? I understand wanting to notify the Ministry if the extension was in error and not valid but I imagine in most cases another extension will be taken right away to correct it. Thoughts?

**Comment by m-prodan** [comment]
I agree with your thoughts on scenario 4 - you're right, if we're deleting it, it's likely to fix an error. So I removed that criteria, and dropped the FOI request piece. Thanks!

**Comment by divyav-aot** [comment]
@arielleandrews @JHarrietha-AOT  @m-prodan - Please confirm if we need to show notification for Approved, Denied and Pending. Also, could you please update the notification message for each AC.

**Comment by m-prodan** [comment]
I think similar to our ideas for comments - we shouldn't have a notification for pending, I will remove that AC.

I'll take a stab at a notification message, feel free to update it @JHarrietha-AOT @arielleandrews

**Comment by arielleandrews** [comment]
I edited the approved notification message in scenario 1, please review @m-prodan and @JHarrietha-AOT. Thanks!

**Comment by JHarrietha-AOT** [comment]
Looks great @arielleandrews!

**Comment by divyav-aot** [comment]
@arielleandrews @JHarrietha-AOT @m-prodan - Do we need to clean previous extension notifications if we have a new approved/denied extension?
For example: JERI-2022-xxxxx has public body extension created. It will show in My Notifications/Watching Notifications. After this, an OIPC extension got approved/denied. Do you want to delete the previous notification (public body) and show only the latest extension notification. Could you please confirm.

**Comment by JHarrietha-AOT** [comment]
@arielleandrews @m-prodan I would think we should just remove any notifications other then approved or denied. After getting the feedback from users in the demo where it was largely people only wanted to see approved or denied extension notifications. 

@divyav-aot  I will leave to for Matt and Arielle to confirm but that is my thoughts

**Comment by divyav-aot** [comment]
@JHarrietha-AOT  - I understand that, we need to show only approved/denied extension. I am confused about the cleanup of previous notifications related to Extension.

**Comment by JHarrietha-AOT** [comment]
@divyav-aot Im thinking all other notifications should be removed. Just have notifications for approved or denied extensions. @m-prodan @arielleandrews would you agree as that is the only ones people gave thumbs up to at that demo?

**Comment by arielleandrews** [comment]
I agree, just approved and denied extensions need notifications. Are the scenarios clear @divyav-aot?

**Comment by divyav-aot** [comment]
@arielleandrews  - I would like to know how the notification should display for below examples.
**Example 1:**
JERI-2022-xxxxx has public body extension created. It will show in My Notifications/Watching Notifications. After this, an OIPC extension got approved/denied. Do you want to delete the previous notification (public body) and show only the latest extension notification. Could you please confirm. 
**Example 2:**
JERI-2022-xxxxx has public body extension created. It will show in My Notifications/Watching Notifications. Then I deleted the extension. Do we need to clean the notification as the extension is not there any more.

**Comment by divyav-aot** [comment]
@arielleandrews @JHarrietha-AOT  - 1 more scenario. There is an approved extension with 2 days. The notification shows "Extension taken for 2 days. The new legislated due date is 2022 Mar 28". If I change the days to 3 and save. Should I show "Extension taken for 3 days. The new legislated due date is 2022 Mar 28". Currently edit of reason and days are not handled.

**Comment by JHarrietha-AOT** [comment]
@arielleandrews this is ready for QA

**Comment by arielleandrews** [comment]
@divyav-aot I added scenarios 4 and 5 in the AC to cover some of the discussion captured in the comments and our meetings. Please let me know if these are accurate. Thanks!

**Comment by divyav-aot** [comment]
Thanks @arielleandrews. All scenarios are captured here.

**Comment by arielleandrews** [comment]
Testing is complete, all scenarios are working as expected. One small change request @divyav-aot: is it possible to have the date display as MMM DD YYYY i.e. Feb 28 2022. We've heard from users this is the preferred date format

**Comment by JHarrietha-AOT** [comment]
Good thought @arielleandrews Just wanted to add as it is not a date picker field we could change it in the table and for the dates at the top of the modal if possible

**Comment by divyav-aot** [comment]
@arielleandrews @JHarrietha-AOT - Date in the Notification message right? It is possible to change. I will take this as low priority as we have few user stories to push.

**Comment by divyav-aot** [comment]
@arielleandrews @JHarrietha-AOT  -  date format change is pushed to test
